sqlobject/src/main/java/org/jdbi/v3/sqlobject/statement/SqlUpdate.java

```diff
@@ -88,11 +85,13 @@ public Handler buildHandler(Class<?> sqlObjectType, Method method) {
         }
 
         @Override
-        public Object invoke(Object target, Object[] args, HandleSupplier handle) {
-            String sql = handle.getConfig(SqlObjects.class).getSqlLocator().locate(sqlObjectType, getMethod());
-            Update update = handle.getHandle().createUpdate(sql);
-            applyCustomizers(update, args);
-            return this.returner.apply(update);
+        Update createStatement(Handle handle, String locatedSql) {
+            return handle.createUpdate(locatedSql);
+        }
+
+        @Override
+        void configureReturner(Update u, SqlObjectStatementConfiguration cfg) {
+            cfg.setReturner(() -> returner.apply(u));
```

Comment:
Feels like the if/else chain in the constructor wants to live in this method.

Maybe `SqlObjectStatementConfiguration.returner` wants to be a `Function<SqlStatement,Object>` instead?